packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts

```diff
@@ -1231,4 +1231,517 @@ describe('changeset', () => {
     expect(differences.resources.differenceCount).toBe(1);
     expect(differences.resources.get('BucketResource').changeImpact === ResourceImpact.WILL_IMPORT);
   });
+
+  test('properties that only show up in changeset diff are included in fullDiff', () => {
+    // GIVEN
+    const currentTemplate = {
+      Parameters: {
+        SsmParameterValuetestbugreportC9: {
+          Type: 'AWS::SSM::Parameter::Value<String>',
+          Default: 'goodJob',
+        },
+      },
+      Resources: {
+        mySsmParameter: {
+          Type: 'AWS::SSM::Parameter',
+          Properties: {
+            Name: 'mySsmParameterFromStack',
+            Type: 'String',
+            Value: {
+              Ref: 'SsmParameterValuetestbugreportC9',
+            },
+          },
+        },
+      },
+    };
+
+    // Template diff misses the change:
+    const diffWithoutChangeSet = fullDiff(currentTemplate, currentTemplate);
+    // THEN
+    expect(diffWithoutChangeSet.differenceCount).toBe(0);
+    expect(diffWithoutChangeSet.resources.changes).toEqual({});
+    // AND THEN
+    const diffWithChangeSet = fullDiff(currentTemplate, currentTemplate,
+      {
+        Changes: [
+          {
+            Type: 'Resource',
+            ResourceChange: {
+              Action: 'Modify',
+              LogicalResourceId: 'mySsmParameter',
+              PhysicalResourceId: 'mySsmParameterFromStack',
+              ResourceType: 'AWS::SSM::Parameter',
+              Replacement: 'False',
+              Scope: ['Properties'],
+              Details: [{
+                Target: { Attribute: 'Properties', Name: 'Value', RequiresRecreation: 'Never' },
+                Evaluation: 'Static',
+                ChangeSource: 'DirectModification',
+              }],
+            },
+          },
+        ],
+        Parameters: [{
+          ParameterKey: 'SsmParameterValuetestbugreportC9',
+          ParameterValue: 'goodJob',
+          ResolvedValue: 'changedVal',
+        }],
+      },
+    );
+
+    // THEN
+    expect(diffWithChangeSet.differenceCount).toBe(1);
+    expect(diffWithChangeSet.resources.changes).toEqual(
```

Comment:
we don't need to assert against all of this, can we just assert against `diffWithChangeSet.resources.changes.mySssmParameter.propertyUpdates`?

Reply:
What's the benefit of asserting less? On my previous team, we always asserted as much as possible information

Reply:
The test assertions are long enough that they're difficult to follow; I have a hard time telling what part of that assertion you care about for this test. There's a lot of noise in each assertion

Reply:
I think that's fair, I'm willing to make that change

Reply:
I agree with @bergjaak, I'd rather that we asserted against more than less.